crates/sui-bridge/src/eth_client.rs

```diff
@@ -58,11 +58,39 @@ where
         Ok(())
     }
 
-    pub async fn get_bridge_events_maybe(
+    pub async fn get_finalized_bridge_action_maybe(
         &self,
-        tx_hash: &str,
-    ) -> BridgeResult<Vec<EthBridgeEvent>> {
-        unimplemented!()
+        tx_hash: TxHash,
+        event_idx: u16,
+    ) -> BridgeResult<BridgeAction> {
+        let receipt = self
+            .provider
+            .get_transaction_receipt(tx_hash)
+            .await
+            .map_err(BridgeError::from)?
+            .ok_or(BridgeError::TxNotFound)?;
+        let receipt_block_num = receipt.block_number.ok_or(BridgeError::ProviderError(
+            "Provider returns log without block_number".into(),
+        ))?;
+        let last_finalzied_block_id = self.get_last_finalized_block_id().await?;
+        if receipt_block_num.as_u64() > last_finalzied_block_id {
```

Comment:
This line would only be true for the period in between the block proposal and until the block receives enough votes. But in the case of network delays or equivocation by validators, where the blockchain's state might temporarily diverge, will we need some additional checks?

https://ethereum.org/en/developers/docs/consensus-mechanisms/pos/#fork-choice

Reply:
"finalized" blocks are considered highly unlikely to be reverted in forks. This simplifies the model the code is written. 

> A finalized block is a justified block that is 1 epoch behind the most recently justified block. Finalized blocks are extremely [unlikely to be re-organized](https://blog.ethereum.org/2021/11/29/how-the-merge-impacts-app-layer), with the only exception being if a two-thirds majority of validators finalize a competing chain of blocks.

https://www.alchemy.com/overviews/ethereum-commitment-levels

But in the event of an actual fork involving a previous finalized block, the code cannot handle it properly ...